packages/lib-classifier/src/plugins/tasks/TextTask/components/TextTask/TextTask.js

```diff
@@ -0,0 +1,75 @@
+import { PlainButton } from '@zooniverse/react-components'
+import { Box, FormField, TextArea } from 'grommet'
+import React from 'react'
+
+function TextTask (props) {
+  const { autoFocus, disabled, task } = props
+  const defaultValue = task.annotation ? task.annotation.value : ''
+  const [value, setValue] = React.useState(defaultValue)
+  const textArea = React.createRef()
+
+  function onChange (event) {
+    const { target } = event
+    updateText(target.value)
+  }
+
+  function updateText (text) {
+    setValue(text)
+    task.updateAnnotation(text)
+  }
+
+  function setTagSelection (e) {
+    const textTag = e.currentTarget.value
+    const startTag = `[${textTag}]`
+    const endTag = `[/${textTag}]`
+    const selectionStart = textArea.current.selectionStart
+    const selectionEnd = textArea.current.selectionEnd
+    const textBefore = value.substring(0, selectionStart)
+    let textAfter
+    let newValue
+
+    if (selectionStart === selectionEnd) {
+      textAfter = value.substring(selectionStart, value.length)
+      newValue = textBefore + startTag + endTag + textAfter
+    } else {
+      const textInBetween = value.substring(selectionStart, selectionEnd)
+      textAfter = value.substring(selectionEnd, value.length)
+      newValue = textBefore + startTag + textInBetween + endTag + textAfter
+    }
+    updateText(newValue)
+    if (textArea.current.focus) {
+      textArea.current.setSelectionRange((newValue.length - textAfter.length), (newValue.length - textAfter.length))
+      textArea.current.focus()
+    }
+  }
+
+  return (
+    <Box>
+      <FormField
+        htmlFor={`${task.taskKey}-${task.type}`}
+        label={task.instruction}
+      >
+        <TextArea
+          ref={textArea}
+          autoFocus={autoFocus}
+          disabled={disabled}
+          id={`${task.taskKey}-${task.type}`}
+          value={value}
+          onChange={onChange}
+        />
+      </FormField>
+      {task.text_tags.map(tag => (
+        <PlainButton
+          key={tag}
+          disabled={disabled}
+          justify='start'
+          onClick={setTagSelection}
+          text={tag}
+          value={tag}
+        />
+      ))}
```

Comment:
Maybe wrap this in a `Box` with the direction set to row for display of multiple buttons if enabled?

Reply:
Thanks! I didn’t know how to style this.